java-checks/src/main/java/org/sonar/java/checks/VolatileNonPrimitiveFieldCheck.java

```diff
@@ -36,6 +36,44 @@
   private static final String REF_MESSAGE = "Remove the \"volatile\" keyword from this field.";
   private static final String ARRAY_MESSAGE = "Use an \"Atomic%sArray\" instead.";
 
+  private static final List<String> STANDARD_IMMUTABLE_TYPES = Arrays.asList(
+    "java.awt.Color",
+    "java.awt.Cursor",
+    "java.awt.Font",
+    "java.io.File",
+    "java.lang.Boolean",
+    "java.lang.Byte",
+    "java.lang.Character",
+    "java.lang.Double",
+    "java.lang.Float",
+    "java.lang.Integer",
+    "java.lang.Long",
+    "java.lang.Short",
+    "java.lang.String",
+    "java.math.BigDecimal",
+    "java.math.BigInteger",
+    "java.net.URL",
+    "java.time.Clock",
+    "java.time.DayOfWeek",
+    "java.time.Instant",
+    "java.time.LocalDate",
+    "java.time.LocalDateTime",
+    "java.time.LocalTime",
+    "java.time.Month",
+    "java.time.MonthDay",
+    "java.time.OffsetDateTime",
+    "java.time.OffsetTime",
+    "java.time.Year",
+    "java.time.YearMonth",
+    "java.time.ZoneId",
+    "java.time.ZoneOffset",
+    "java.time.ZonedDateTime",
+    "java.time.Duration",
+    "java.time.Period",
+    "java.util.Date",
```

Comment:
`java.util.Date` is not immutable, see https://docs.oracle.com/javase/7/docs/api/java/util/Date.html#setDate(int)

Reply:
Indeed, nice catch, I have updated the ticket aswell!